### PR TITLE
fixes #415 added two integration test and fixed 4 bugs

### LINKED ISSUE #415: Try to reproduce left over data columns
In #280 I posted a comment where I saw left over data columns.  Need to try to reproduce this with an IT.

### PR DESCRIPTION
**Pull request by keith-turner**
Started off adding CollisionIT and found 4 bugs as a result.  Wrote more specific IT for two of the bugs found.

### CODE REVIEW

**Comment by keith-turner** [comment]
Some of the changes in the patch are best viewed ignoring white-space changes.  I removed a try block.

https://github.com/blog/967-github-secrets

File: modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        loader.execute(new NumLoader(num));
+      }
+    }
+
+    miniFluo.waitForObservers();
+
+    try (TypedSnapshot snapshot = typeLayer.wrap(client.newSnapshot())) {
+
+      for (int i = 0; i < expectedCounts.length; i++) {
+        Assert.assertEquals(expectedCounts[i], snapshot.get().row(i).fam("stat").qual("total").toInteger(-1));
```
Comment (mikewalch) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: This is very cool test.  How about about also checking checking `processed` here?

Comment (keith-turner) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: oh yeah, the more the test verifies the better

File: modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    Scanner scanner = conn.createScanner(getCurTableName(), Authorizations.EMPTY);
+
+    HashSet<String> rowCols = new HashSet<>();
+
+    for (Entry<Key,Value> entry : scanner) {
+      Key k = entry.getKey();
+      String rowCol = k.getRow() + ":" + k.getColumnFamily() + ":" + k.getColumnQualifier() + ":"
+          + String.format("%x", k.getTimestamp() & ColumnConstants.PREFIX_MASK);
+      Assert.assertFalse("Duplicate row col " + rowCol, rowCols.contains(rowCol));
```
Comment (mikewalch) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: Interesting.. How can garbage collection cause duplicate row columns?  Just curious...

Comment (keith-turner) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: There should be duplicates before GC.  Expecting no duplicates after.

File: modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java (hunk 1 above)
Comment (keith-turner) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: added in 361e076

File: modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java (hunk 2 above)
Comment (keith-turner) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: If you change the test and add calls to `printTable()` (defined in `ITBase`) before and after compaction, should see the duplicates

Comment (mikewalch) [inline modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java]: Makes sense.  I will add add the calls to `printTable()` to check it out.

**Comment by mikewalch** [comment]
+1

**Pull request merged**